airflow/executors/mesos_executor.py

```diff
@@ -0,0 +1,468 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import threading
+from builtins import str
+from queue import Queue
+from typing import Any, Optional
+
+from avmesos.client import MesosClient
+
+from airflow.configuration import conf
+from airflow.exceptions import AirflowException
+from airflow.executors.base_executor import BaseExecutor, CommandType
+from airflow.models.taskinstance import TaskInstance, TaskInstanceKey
+from airflow.utils.session import provide_session
+from airflow.utils.state import State
+
+DEFAULT_FRAMEWORK_NAME = 'Airflow'
+FRAMEWORK_CONNID_PREFIX = 'mesos_framework_'
+
+
+def get_framework_name():
+    """Get the mesos framework name if its set in airflow.cfg"""
+    if not conf.get('mesos', 'FRAMEWORK_NAME'):
+        return DEFAULT_FRAMEWORK_NAME
+    return conf.get('mesos', 'FRAMEWORK_NAME')
+
+
+# pylint: disable=too-many-nested-blocks
+# pylint: disable=too-many-instance-attributes
+class AirflowMesosScheduler(MesosClient):
+    """
+    Airflow Mesos scheduler implements mesos scheduler interface
+    to schedule airflow tasks on mesos
+    Basically, it schedules a command like
+    'airflow run <dag_id> <task_instance_id> <start_date> --local -p=<pickle>'
+    to run on a mesos slave
+    """
+
+    # pylint: disable=super-init-not-called
+    def __init__(self,
+                 executor,
+                 task_queue,
+                 result_queue,
+                 task_cpu=1,
+                 task_mem=256):
```

Comment:
Please add type hints, we are using mypy and there's a big work on adding them everywhere in the codebase

Reply:
ok, I will. :-)

Reply:
How to type hint a `Queue()` object like task_queue? :man_shrugging: